File: src/main/java/com/github/ericjgagnon/vitest/run/VitestConfigurationType.java
package com.github.ericjgagnon.vitest.run;

import com.intellij.execution.configurations.ConfigurationTypeUtil;
import com.intellij.execution.configurations.RunConfigurationSingletonPolicy;
import com.intellij.execution.configurations.SimpleConfigurationType;
import com.intellij.icons.AllIcons;
import com.intellij.openapi.project.DumbAware;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.util.NotNullLazyValue;
import org.jetbrains.annotations.NotNull;

public class VitestConfigurationType extends SimpleConfigurationType implements DumbAware {
    protected VitestConfigurationType() {
        super("JavaScriptTestRunnerVitest", "Vitest", null, NotNullLazyValue.createValue(() -> {
            return AllIcons.FileTypes.JavaScript;
        }));
    }

    @Override
    public @NotNull VitestRunConfiguration createTemplateConfiguration(@NotNull Project project) {
        return new VitestRunConfiguration(project, this, "Vitest");
    }

    @NotNull
    public static VitestConfigurationType getInstance() {
        return ConfigurationTypeUtil.findConfigurationType(VitestConfigurationType.class);
    }

    @Override
    public @NotNull RunConfigurationSingletonPolicy getSingletonPolicy() {
        return RunConfigurationSingletonPolicy.SINGLE_INSTANCE_ONLY;
    }
}

File: src/main/kotlin/com/github/ericjgagnon/vitest/run/VitestConfigurationType.kt
package com.github.ericjgagnon.vitest.run

import com.github.ericjgagnon.vitest.run.VitestConstants.CONFIG_TYPE_NAME
import com.github.ericjgagnon.vitest.run.VitestConstants.TEST_FRAMEWORK_NAME
import com.intellij.execution.configurations.ConfigurationTypeUtil.findConfigurationType
import com.intellij.execution.configurations.RunConfigurationSingletonPolicy
import com.intellij.execution.configurations.SimpleConfigurationType
import com.intellij.openapi.project.DumbAware
import com.intellij.openapi.project.Project
import com.intellij.openapi.util.NotNullLazyValue
import icons.Icons

class VitestConfigurationType: SimpleConfigurationType(TEST_FRAMEWORK_NAME,
    CONFIG_TYPE_NAME,
    null,
    NotNullLazyValue.createValue { Icons.VITEST_ICON }), DumbAware {
    override fun createTemplateConfiguration(project: Project): VitestRunConfiguration {
        return VitestRunConfiguration(project, this, CONFIG_TYPE_NAME)
    }

    override fun getSingletonPolicy(): RunConfigurationSingletonPolicy {
        return RunConfigurationSingletonPolicy.SINGLE_INSTANCE_ONLY
    }

    companion object {
        @JvmStatic
        val instance: VitestConfigurationType
            get() = findConfigurationType(VitestConfigurationType::class.java)
    }
}